miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditServiceRecordController.java

```diff
@@ -0,0 +1,147 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import org.springframework.web.bind.support.SessionStatus;
+import org.springframework.web.servlet.ModelAndView;
+
+import uk.ac.bbsrc.tgac.miso.core.data.AbstractSequencerServiceRecord;
+import uk.ac.bbsrc.tgac.miso.core.data.SequencerReference;
+import uk.ac.bbsrc.tgac.miso.core.data.SequencerServiceRecord;
+import uk.ac.bbsrc.tgac.miso.core.factory.DataObjectFactory;
+import uk.ac.bbsrc.tgac.miso.core.manager.FilesManager;
+import uk.ac.bbsrc.tgac.miso.core.manager.RequestManager;
+
+@Controller
+@RequestMapping("/stats/sequencer/servicerecord")
+@SessionAttributes("serviceRecord")
+public class EditServiceRecordController {
+  
+  protected static final Logger log = LoggerFactory.getLogger(EditServiceRecordController.class);
+  
+  @Autowired
+  private RequestManager requestManager;
+
+  @Autowired
+  private FilesManager filesManager;
+
+  @Autowired
+  private DataObjectFactory dataObjectFactory;
+
+  public void setRequestManager(RequestManager requestManager) {
+    this.requestManager = requestManager;
+  }
+
+  public void setFilesManager(FilesManager filesManager) {
+    this.filesManager = filesManager;
+  }
+
+  public void setDataObjectFactory(DataObjectFactory dataObjectFactory) {
+    this.dataObjectFactory = dataObjectFactory;
+  }
+  
+  public Map<Integer, String> populateServiceRecordFiles(SequencerServiceRecord record) throws IOException {
+    if (record.getId() != AbstractSequencerServiceRecord.UNSAVED_ID) {
+      Map<Integer, String> fileMap = new HashMap<Integer, String>();
+      for (String s : filesManager.getFileNames(SequencerServiceRecord.class, String.valueOf(record.getId()))) {
+        fileMap.put(s.hashCode(), s);
+      }
+      return fileMap;
+    }
+    return Collections.emptyMap();
+  }
+  
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> maxLengths() throws IOException {
+    return requestManager.getServiceRecordColumnSizes();
+  }
+
+  @RequestMapping(value = "/{recordId}", method = RequestMethod.GET)
+  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, ModelMap model) throws IOException {
+    SequencerServiceRecord sr = requestManager.getSequencerServiceRecordById(recordId);
+    if (sr != null) {
+      model.put("serviceRecord", sr);
+      model.put("serviceRecordFiles", populateServiceRecordFiles(sr));
+    } else {
+      throw new IOException("Cannot retrieve the requested Service Record");
```

Comment:
Currently, most controllers throw a SecurityException (unchecked) if the object isn't found (requestManager call returns null). I was going for consistency, but used IOException (checked) instead as it makes a bit more sense. The Request Manager throws an IOException if there is a permission issue, so a return of null just means the object doesn't exist (not a security issue). The result is the same: error is logged, user sees a message + stack trace.

I agree that it's not the best user experience, and should be changed in all Controller classes at some point. A 404 page (and no logging since it's a user error) would probably be best. Would you prefer that I do this (or something else) here, or create a separate ticket to address this in all controllers including this one at some point later on?